proto/store/setting.proto

```diff
@@ -33,6 +33,27 @@ message WorkspaceProfileSetting {
 
   // The duration for refresh token.
   google.protobuf.Duration refresh_token_duration = 6;
+
+  // The setting of custom announcement
+  AnnouncementSetting announcement_setting = 7;
+}
+
+message AnnouncementSetting {
+  // We support four levels of AlertLevel: INFO, WARNING, and ERROR.
+  enum AnnouncementAlertLevel {
+    ANNOUNCEMENTALERTLEVEL_INFO = 0;
+    ANNOUNCEMENTALERTLEVEL_WARNING = 1;
+    ANNOUNCEMENTALERTLEVEL_ERROR = 2;
+  }
+
+  // The alert level of announcemnt
+  AnnouncementAlertLevel announcement_level = 1;
+
+  // The text of announcemnt
+  string announcement_text = 2;
+
+  // The optional link, user can follow the link to check extra details
+  string announcement_link = 3;
 }
```

Comment:
Since these are already in the `AnnouncementSetting` message, we may remove the `announcement` prefix from the field names.